### PR TITLE
feat: add showCheckedStrategy api  for cascader

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/34193

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Cascader supports `showCheckedStrategy ` for value display strategy.      |
| 🇨🇳 Chinese |  Cascader  添加 `showCheckedStrategy` 用于配置回填方式。    |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/cascader/index.zh-CN.md [hunk 1]
```diff
@@ -55,6 +55,7 @@ cover: https://gw.alipayobjects.com/zos/alicdn/UdS8y8xyZ/Cascader.svg
 | onChange | 选择完成后的回调 | (value, selectedOptions) => void | - |  |
 | onDropdownVisibleChange | 显示/隐藏浮层的回调 | (value) => void | - | 4.17.0 |
 | multiple | 支持多选节点 | boolean | - | 4.17.0 |
+| showCheckedStrategy | 配置 `multiple`模式 时，定义选中项回填的方式。`Cascader.SHOW_CHILD`: 只显示选中的子节点。`Cascader.SHOW_PARENT`: 只显示父节点(当父节点下所有子节点都选中时)。 | `Cascader.SHOW_PARENT` \| `Cascader.SHOW_CHILD` | `Cascader.SHOW_PARENT` | 4.20.0 |
```
Comment (afc163) [inline components/cascader/index.zh-CN.md:58]:
```suggestion
| showCheckedStrategy | 配置 `multiple` 模式 时，定义选中项回填的方式。`Cascader.SHOW_CHILD`: 只显示选中的子节点。`Cascader.SHOW_PARENT`: 只显示父节点（当父节点下所有子节点都选中时）。 | `Cascader.SHOW_PARENT` \| `Cascader.SHOW_CHILD` | `Cascader.SHOW_PARENT` | 4.20.0 |
```

Comment (heiyu4585) [reply components/cascader/index.zh-CN.md]: done

File: components/cascader/demo/multiple.md [hunk 2]
```diff
@@ -62,6 +64,7 @@ const App = () => {
       onChange={onChange}
       multiple
       maxTagCount="responsive"
+      showCheckedStrategy={SHOW_CHILD}
```
Comment (afc163) [inline components/cascader/demo/multiple.md:67]: 多选 demo 默认不加，可以新增 demo 演示这三种情况的区别。

Comment (heiyu4585) [reply components/cascader/demo/multiple.md:67]: 默认是`PARENT`,  是不是只演示`CHILD` 和 `PARENT`两种情况就行

Comment (afc163) [reply components/cascader/demo/multiple.md:67]: 可以

Comment (heiyu4585) [reply components/cascader/demo/multiple.md:67]: done

File: components/cascader/demo/showCheckedStrategy.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+order: 5.1
+version: 4.20.0
+title:
+  zh-CN: 自定义回填方式
+  en-US: ShowCheckedStrategy
+---
+
+## zh-CN
+
+通过设置`ShowCheckedStrategy`选择回填方式。
```
Comment (afc163) [inline components/cascader/demo/showCheckedStrategy.md:11]:
```suggestion
通过设置 `ShowCheckedStrategy` 选择回填方式。
```

File: components/cascader/demo/showCheckedStrategy.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+## en-US
+
+The way show selected item in box when `ShowCheckedStrategy` set
```
Comment (afc163) [inline components/cascader/demo/showCheckedStrategy.md:15]:
```suggestion
The way show selected item in box using `ShowCheckedStrategy`.
```

File: components/cascader/demo/showCheckedStrategy.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+      />
+      <br />
+      <br />
+      <Cascader
+        style={{ width: '100%' }}
+        options={options}
+        onChange={onChange}
+        multiple
+        maxTagCount="responsive"
+        showCheckedStrategy={SHOW_PARENT}
```
Comment (afc163) [inline components/cascader/demo/showCheckedStrategy.md:83]: 默认选型不用写吧。

Comment (heiyu4585) [reply components/cascader/demo/showCheckedStrategy.md]: done

File: components/cascader/demo/showCheckedStrategy.md (hunk 4 above)
Comment (heiyu4585) [reply components/cascader/demo/showCheckedStrategy.md]: done

**Review by afc163 — APPROVED** [review]

File: components/cascader/demo/multiple.md (hunk 2 above)
Comment (afc163) [reply components/cascader/demo/multiple.md:67]: 这个不要改。

Comment (heiyu4585) [reply components/cascader/demo/multiple.md]: done

**Pull request merged**